utils/data_helpers.py

```diff
@@ -282,22 +328,25 @@ def encode_tabular_data(tab_xray_df):
     for column in tab_xray_df:
         if column not in ignores and column not in normalises:
             # One-hot encode the categorical data
-            one_hot_subset_df = pd.get_dummies(tab_xray_df[column])
-            one_hot_subset_df = one_hot_subset_df.add_prefix(f'{column}_')
+            one_hot_subset_df = pd.get_dummies(tab_xray_df[column], prefix=column)
+            # one_hot_subset_df = one_hot_subset_df.add_prefix(f'{column}_')
             encoded_df = encoded_df.join(one_hot_subset_df)
         if column in normalises and column not in ignores:
             # Continious variable, simply normalise
             norm_values = normalise(tab_xray_df[column].values)
             encoded_df = encoded_df.join(pd.DataFrame({column: norm_values}))
-
+    encoded_df['Path'] = encoded_df['Path'].apply(lambda x: remove_prefix(x, prefix=image_path_prefix))
     return encoded_df
 
-def concat_columns_into_vector(encoded_tabular_df):
-    """ Concatenate the values for all features to form an array.
-        We then pair each array / vector with the corresponding image in
-        a dictionary for easy look up.
+def remove_prefix(name, prefix: str):
+    return name[len(prefix):]
+
+def listify(encoded_tabular_df, prefix):
```

Comment:
possibly change the function name? Sounds super generic but is quite specific

Reply:
Any ideas? I struggled to come up with a decent name

Reply:
extract_xray_training_data() or something along those lines? I mean, that name sucks but at least you know what is going on haha

Can just add in the comment that the returns are lists

Reply:
OK, I've come up with something similar :+1: